v2/internal/reconcilers/arm/arm_client.go

```diff
@@ -51,7 +47,7 @@ type Connection struct {
 func newConnection(client *armClient) *Connection {
```

Comment:
This method is called inside `armreconciler.ARMClientFactory` and returns a `Connection` instead of ARMClient. Maybe we need to fix some naming around here for `Connection`.

Reply:
I see what you mean.

I've changed a few things here:

1. `Connection` is now an interface which `armClient` implements, so there's not two separate types.
2. The method is `GetConnection` now rather than `GetClient`, and it returns a `Connection`. This seemed clearer than `GetClient` as there is already such a thing as a client (i.e. `GenericARMClient`) and this was "bigger" than that as it also has identity/credential details on it.